src/sparseml/modifiers/quantization/gptq/pytorch.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Dict, Iterable, List, Optional, Tuple
+
+import torch
+
+from sparseml.core.model import ModifiableModel
+from sparseml.core.state import State
+from sparseml.modifiers.quantization.gptq.base import GPTQModifier
+from sparseml.modifiers.quantization.gptq.utils.gptq_wrapper import GPTQWrapper
+from sparseml.modifiers.utils.layer_compressor import LayerCompressor
+from sparseml.modifiers.utils.pytorch_helpers import run_calibration_forward
+
+
+__all__ = ["GPTQModifierPyTorch"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GPTQModifierPyTorch(GPTQModifier):
+    """
+    Pytorch implementation of SparseGPT
+
+    Lifecycle:
+        - on_initialize
+            - initialize_compression()
+                - compressible_layers()
+                - LayerCompressor.pre_compress()
+            - apply_compression()
+                - run_calibration_forward()
+                - LayerCompressor.compress()
+                - LayerCompressor.post_compress()
+                - LayerCompressor.revert_layer_wrappers()
+
+    | Sample yaml:
+    |   test_stage:
+    |       obcq_modifiers:
+    |           GPTQModifier:
+    |               sequential_update: True
+    |               dampening_frac: 0.001
+    |               targets: __ALL__
```

Comment:
the default here should be to leave the targets blank actually, we infer the correct targets (transformer layers) from the model. Not sure if this sample is in sgpt but we should change it there as well